exporter/indexstats_collector.go

```diff
@@ -41,26 +41,27 @@ func (d *indexstatsCollector) Describe(ch chan<- *prometheus.Desc) {
 }
 
 func (d *indexstatsCollector) Collect(ch chan<- prometheus.Metric) {
+	collections := d.collections
+
 	if d.discoveringMode {
-		databases := map[string][]string{}
-		for _, dbCollection := range d.collections {
-			parts := strings.Split(dbCollection, ".")
-			if _, ok := databases[parts[0]]; !ok {
-				db := parts[0]
-				databases[db], _ = d.client.Database(parts[0]).ListCollectionNames(d.ctx, bson.D{})
-			}
+		namespaces, err := listAllCollections(d.ctx, d.client, d.collections)
```

Comment:
d.collections is used as regex pattern. Is this expectation correct from input perspective?

Reply:
Yes, it is a filter